test/03_USM.test.js

```diff
@@ -29,127 +43,280 @@ contract('USM', (accounts) => {
+        // be 1 (current ETH pool) / 0.004 * (1 - 1 / 2) = 125, ie, the user's FUM balance should increase by a factor of 3/2:
+        fumBalance3.toString().should.equal(wadMul(oneEth, priceWAD).mul(THREE).div(TWO).toString())
+
+        const fundDefundAdj3 = (await usm.fundDefundAdjustment())
+        // We've 2x'd the ETH pool, which according to the adjustment math in USM.fund(), should 4x the fundDefundAdjustment:
+        const targetFundDefundAdj3 = WAD.mul(FOUR)
+        fundDefundAdj3.toString().should.equal(targetFundDefundAdj3.toString())
+
+        const fumBuyPrice3 = (await usm.fumPrice(sides.BUY))
+        const fumSellPrice3 = (await usm.fumPrice(sides.SELL))
+        // The FUM buy & sell prices will have changed now, because the sliding-price above will have effectively collected fees
+        // in the contract, which go to FUM holders.  Calculating the increase would be a bit of work, but one simple check we can
+        // apply is, fundDefundAdjustment = 4 should mean the FUM buy price is now 4x the FUM sell price:
+        fumBuyPrice3.toString().should.equal(wadMul(fumSellPrice3, fundDefundAdj3).toString())
+
+        // Check that fundDefundAdjustment decays properly (well, approximately) over time:
+        // - Our adjustment was previously 4: see targetFundDefundAdj3 above.
+        // - Use timeMachine to move forward 180 secs = 3 min.
+        // - Since USM.BUY_SELL_ADJUSTMENTS_HALF_LIFE = 60 (1 min), 3 min should mean a decay of ~1/8.
+        // - The way our "time decay towards 1" approximation works, decaying 4 by ~1/8 should yield 1 + (4 * 1/8) - 1/8 = 1.375.
+        //   (See comment in USM.mintBurnAdjustment().)
+        // - And finally - since our decay's exponentiation is approximate (WadMath.wadHalfExp()), don't compare fundDefundAdj4 to
+        //   targetFundDefundAdj4 *exactly*, but instead first round both to the nearest multiple of 10 (via divRound(10)).
+        const block0 = (await web3.eth.getBlockNumber())
+        const t0 = (await web3.eth.getBlock(block0)).timestamp
+        const timeDelay = 3 * 60
+        await timeMachine.advanceTimeAndBlock(timeDelay)
+        const block1 = (await web3.eth.getBlockNumber())
+        const t1 = (await web3.eth.getBlock(block1)).timestamp
+        t1.toString().should.equal((t0 + timeDelay).toString())
+        const fundDefundAdj4 = (await usm.fundDefundAdjustment())
+        const decayFactor4 = WAD.div(EIGHT)
+        const targetFundDefundAdj4 = WAD.add(wadMul(fundDefundAdj3, decayFactor4)).sub(decayFactor4)
+        fundDefundAdj4.divRound(TEN).toString().should.equal(targetFundDefundAdj4.divRound(TEN).toString())
       })
 
-      describe('with existing FUM supply', () => {
+      describe("with existing FUM supply", () => {
         beforeEach(async () => {
           await usm.fund(user1, user1, oneEth, { from: user1 })
```

Comment:
Should we use `ethPool()` instead of `oneEth`, to parameterize?

Reply:
I don't think these parameterization requests should be too hard but enough for today.  Aiming to pick 'em up in the next couple days

Reply:
I made some changes to test 3 to further parameterize it, a little more painful than I expected.  We could go further but maybe that's enough until we're actually putting it through random inputs.